array_analyzer/extract/txt_parser.py

```diff
@@ -184,3 +289,20 @@ def populate_array_antigen(arr, id_arr_, repl):
 
     return arr
 
+
+def populate_array_antigen(arr, csv_antigens_):
+    """
+    populates an array with antigen
+        used for metadata obtained through .csv files
+    :param arr:
+    :param csv_antigens_: list
+        list of dictionaries whose keys define array coordinates and values
+    :return:
+    """
+    for antigen in csv_antigens_:
+        r = antigen['@row']
+        c = antigen['@col']
+        v = antigen['@antigen']
+        arr[r, c] = v
+
+    return arr
```

Comment:
There are many similar `populate_array_...` methods in this module to covert dictionaries into 2D arrays. Maybe consider to consolidate them in the next refactor if not in this PR. I believe you can achieve the same thing in 2 lines with pandas:
https://stackoverflow.com/questions/20638006/convert-list-of-dictionaries-to-a-pandas-dataframe
https://stackoverflow.com/questions/9339184/python-pandas-dataframe-convert-1column-data-into-2d-data-format

Reply:
I thought about this more and tried something.  It's the new function txt_parser.create_xlsx_array.  It doesn't work right now, but could probably be fixed later on.  Let's revisit after this PR.